### ISSUE TITLE
Provide a method to resume (non-prebinded) XMPP sessions after page reload

### ISSUE THREAD

**Issue by thomascube** [jcbrand/converse.js#155]
The primary way of keeping sessions alive throughout multiple pages or page reloads is to use pre-binding on the server side and then provide JID, SID and RID to the converse instance. This only works when the server knows all the user credentials in order to do the pre-binding. But there are also use-cases where users login to XMPP directly from converse.js. Therefore it would be nice to have a way to advice converse.js to keep session alive after page loads. I admit that storing all session data in sessionStorage or localStorage may raise some security concerns as it would allow other scripts with access to the browser's sessionStorage to easily hijack sessions.

For proper design, the SID and JID could be stored elsewhere by the website, preferably in a server-side session container. This part can already be implemented using converse's events:
```js
converse.on('onReady', function(e){
    var data = { jid: e.target.bare_jid, sid: converse.getSID() };
    // send data to server for storing in session
});
```

What's missing is a method (or configuration option) for the application embedding converse.js to pass these two values again to resume a previous session. The RID associated with the SID should be kept in sessionStorage by converse.js itself and be restored directly:
```js
converse.initialize({
    resume: true,
    jid: data.jid,  // from server-side session data
    sid: data.sid   // from server-side session data
});
```

**Comment by chihanlin** [comment]
we have used the 0.8.6, and found even though the sending/receiving  users are showed on line, but, when try to use converse.js web client to chat with the online user, the other side of user has never receives the message. I suspect that the sender web xmpp client has lost its connection session. (but from web interface, it show on line.) we have to manually refresh the page, or reload the page to make the session connected. (we found this work around.) I supposed, this may relate this topic.  Any way to resolve this issue?
